core/wiki/macros/toc.tid

```diff
@@ -2,182 +2,214 @@ title: $:/core/macros/toc
 tags: $:/tags/Macro
 
 \define toc-caption()
-<$set name="tv-wikilinks" value="no">
-  <$transclude field="caption">
-    <$view field="title"/>
-  </$transclude>
-</$set>
+\whitespace trim
+<$set name="tv-wikilinks" value=no>
+  <$list filter="[all[current]has[caption]]" 
+      emptyMessage="""<$transclude field=title><$view field=title/></$transclude>""">
```

Comment:
It's wrong to transclude the `title` field; it should only ever be displayed as plain text.

In any case, I don't think I quite understand what the changes to `toc-caption` are trying to accomplish.

Reply:
If the `caption` field exists, but it is empty, it should use the tiddler title.